### ISSUE TITLE
FFT strange behavoir with HackRF source

### ISSUE THREAD

**Issue by ngmacha** [AlexandreRouma/SDRPlusPlus#705]
Hello! Thank you for a nice software! I've found some strange bug happening whne the the Source is using HackRF. When HackRF is selected as the Source, the FFT windows shows higher frequencies on the left and lower on the rights, in opposite direction with the frequency scale below it. So when I drag the frequency scale below the FFT right-to-left I see the VFO frequency increases, but the FFT is moving in opposite direction, left-to-right! If the Tuning Mode is Normal, clicking somewhere within the FFT then Frequency Selector shows totally wrong frequency, not the VFO tuned frequency. If the Tuning Mode is Center Tuning, clicking FFT jumps the VFO to the frequency shown on the scale, but not corresponding to the FFT/waterfalls shown on screen, with the Frequency Selector correctly show the VFO frequency, but otherwise the FFT is left-right mirrored at the center of VFO. 
![SDRPP_FFTissue_HackRFsource](https://user-images.githubusercontent.com/883534/160434582-e9125ab7-00ec-45af-b567-4429bec7e178.jpg)
This happens only with HackRF source, with RTL-SDR everything behaves normally - I don't have other sources to test though.
Hope I explained the issue understandable enough to fix this. Thank you again for a nice software, otherwise.

**Comment by AlexandreRouma** [comment]
your gain is too high, the hackrf is clipping and te stuff moving the wrong way around are called images. The HackRF is especially prone to this.

**Comment by ngmacha** [comment]
> your gain is too high, the hackrf is clipping and te stuff moving the wrong way around are called images. The HackRF is especially prone to this.

Oh, thank you very much for pointing that out!! I used my HackRF some long time ago with the Ham-It-Up for SWL on SDR#, connected it today to test and compare with RTL-SDR using your SDR++ and didn't notice the gain was all the way up as for the RTL-SDR!! You are right, everything is normal on shortwave, just the FM broadcast is too strong that causes HackRF to clip! Never noticed that before, silly me, such an embarrassment! Trying your SDR++ further :-)
Closing this now.
